docs/FormatsAndConventions.md

```diff
@@ -32,20 +32,94 @@ Commit messages for non-trivial commits should have a _**body**_ giving details
  
 1. Separate subject from body with a blank line
 1. Wrap the body at 72 characters
-1. Use the body to explain what and why vs. how
-
-Here is an example (adapted from [this commit](https://github.com/CS2103AUG2016-T11-C4/main/pull/2/commits/5c5b12d1a75c4a73a8330bfd05e406233694ffa3)):
- 
- ```
- Group overloaded methods together
- 
- To make it easier to see the different ways a method can be called, 
- we should group them together.
- 
- Note that this is not required strictly by the coding standard.
- ```
+1. Use the body to explain WHAT the commit is about and WHY it was done that way. The reader can refer to the diff to 
+   understand HOW the change was done.
   
-Refer to the article _[How to Write a Git Commit Message](http://chris.beams.io/posts/git-commit/)_ for a more detailed explanation.
+   * Give an explanation for the change(s) that is detailed enough so that the reader can judge if it is a
+     good thing to do, without reading the actual diff to determine how well the code does what the explanation 
+     promises to do.<br>
```

Comment:
Take this example:
* If the commit message says something like `let's use observer pattern to decouple ui from logic`. 

  (a) The reader can immediately say, "no, that's not the right thing to do; the underlying framework already decouples the ui from logic". There is no need to read the diff.
  (b) The reader can agree with the approach, proceed to read the diff, and find that the application of observer pattern is not implemented correctly.

* If the commit message says `let's modify the design of logic a bit` (to give an extreme example), the reader _has_ to read the diff before deciding whether it is the right thing to do and if it is done right. 

It's hard to draw an exact line to say what is `too much/little details in commit message` but the idea is to allow the reader make a high-level set of conclusions (e.g. if the approach taken by the commit agrees with the project philosophy) by reading the commit message before having to read the diff. In some cases, this will spare the reader from having to read the diff at all. Perhaps another way to explain it is, the commit message can explain WHAT the commit message is doing and the diff can show HOW the commit is doing it (much like the header comment vs method body).

Does that answer your question?